creator-node/src/premiumContent/premiumContentAccessChecker.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  isCIDForPremiumTrack,
+  isPremiumContentMatch,
+  isRegisteredDiscoveryNode
+} from './helpers'
+import {
+  CheckAccessArgs,
+  CheckAccessResponse,
+  PremiumContentAccessError
+} from './types'
+import { recoverWallet } from '../apiSigning'
+
+export interface AccessChecker {
+  checkPremiumContentAccess(args: CheckAccessArgs): Promise<CheckAccessResponse>
+}
+
+export class PremiumContentAccessChecker implements AccessChecker {
```

Comment:
actually since this function is only used inside `premiumContentMiddleware()`, we can just move it to be an internal function inside that middleware file

Reply:
agree too, but it allows for easier stubbing for testing purposes

Reply:
ah right, makes sense to export then